ledger-core/virtual/integration/statereport_test.go

```diff
@@ -179,3 +183,69 @@ func TestVirtual_VStateReport_BadState_StateAlreadyExists(t *testing.T) {
 
 	checkBalance(ctx, t, server, objectGlobal, testBalance)
 }
+
+func TestVirtual_VStateReport_CheckValidatedState(t *testing.T) {
+	t.Log("C5124")
+	t.Skip("https://insolar.atlassian.net/browse/PLAT-402")
+
+	mc := minimock.NewController(t)
+
+	server, ctx := utils.NewServer(nil, t)
+	defer server.Stop()
+
+	server.IncrementPulseAndWaitIdle(ctx)
+
+	var (
+		objectLocal          = server.RandomLocalWithPulse()
+		objectGlobal         = reference.NewSelf(objectLocal)
+		dirtyWalletState     = makeRawWalletState(initialBalance)
+		validatedWalletState = makeRawWalletState(initialBalance + 123)
+	)
+
+	content := &payload.VStateReport_ProvidedContentBody{
+		LatestDirtyState: &payload.ObjectState{
+			Reference: server.RandomLocalWithPulse(),
+			Class:     testwalletProxy.GetClass(),
+			State:     dirtyWalletState,
+		},
+		LatestValidatedState: &payload.ObjectState{
+			Reference: server.RandomLocalWithPulse(),
+			Class:     testwalletProxy.GetClass(),
+			State:     validatedWalletState,
+		},
+	}
+
+	{
+		payload := &payload.VStateReport{
+			Status:          payload.Ready,
+			Object:          objectGlobal,
+			ProvidedContent: content,
+		}
+
+		wait := server.Journal.WaitStopOf(&handlers.SMVStateReport{}, 1)
+		server.SendPayload(ctx, payload)
+		waitWithTimeout(wait)
+	}
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+	typedChecker.VStateReport.Set(func(report *payload.VStateReport) bool {
+		require.Equal(t, report.ProvidedContent.LatestDirtyState, content.LatestDirtyState)
+		require.Equal(t, report.ProvidedContent.LatestValidatedState, content.LatestDirtyState)
+		return true
+	})
+
+	wait := server.Journal.WaitStopOf(&handlers.SMVStateReport{}, 1)
+	server.IncrementPulseAndWaitIdle(ctx)
+	waitWithTimeout(wait)
```

Comment:
whats wrong with testutils.WaitSignalsTimed?

Reply:
yes, I forget about it